app/services/market_history_dedupe_service.rb

```diff
@@ -34,6 +34,7 @@ def dedupe(data, server_id)
     if REDIS[server_id].get("HISTORY_RECORD_SHA256:#{sha256}").nil?
       REDIS[server_id].set("HISTORY_RECORD_SHA256:#{sha256}", '1', ex: 600)
       return if data['AlbionId'] == 0 # sometimes the client sends us 0 for the numeric item id, we trash this data
+      return if data['LocationId'] == 0 || !data['LocationId'].is_a?(Numeric)  # sometimes the client sends us 0 for the numeric location id, or a string, we trash this data
```

Comment:
there should be a test for this, testing a string as location id. example of testing location=0 https://github.com/ao-data/albiondata-server-rails/blob/main/spec/app/services/market_history_dedupe_service_spec.rb#L14

Reply:
Done